types/react/index.d.ts

```diff
@@ -334,7 +334,7 @@ declare namespace React {
         displayName?: string;
     }
     function createContext<T>(
-        defaultValue: T,
+        defaultValue?: T,
```

Comment:
This means you could write `createContext<object>();` and get no compile errors when accessing the context unguarded.

What we want is probably an overload. Something like
```ts
function createContext<T>(): Context<T | undefined>;
```

Reply:
Looks good to me, should I apply the suggested change `function createContext<T>(): Context<T | undefined>;` and we can LGTM?

Reply:
Sure. But don't forget to add a test.